### DISCUSSION TITLE
Background message handling doesn't work

### DISCUSSION THREAD

**Discussion by NaviWojciech** [firebase/flutterfire#4805] [Q&A]
Hello!

I use:
```
  firebase_messaging: ^8.0.0-dev.14
  firebase_core: ^0.7.0
```

I try to add handling of background messages to my app. This is a snippet from the code:

```
    FirebaseMessaging.onBackgroundMessage(
      (message) => _handleBackgroundMessage(
          message,
          widget.handler?.onBackgroundMessage,
          NotificationType.backgroundMessage),
    );
```

where:
```
Future<void> _handleBackgroundMessage(
  RemoteMessage message,
  MessageListener listener,
  NotificationType type,
) async {
  print("HANDLE MESSAGE");
}
```
it is a top-level function.

However, when I try to handle background notification, the following message is logged:
_W/FLTFireMsgService(28069): A background message could not be handled in Dart as no onBackgroundMessage handler has been registered._

Am I missing anything there?

**Comment by NaviWojciech** [comment]
Hello! Any ideas?

**Reply by Anan-Saadi** [reply]
did you find any solution? I am facing the same issue
